python/lsst/ci/hsc/validate.py

```diff
@@ -6,6 +6,16 @@
 import functools
 from lsst.pex.logging import getDefaultLog
 from lsst.daf.persistence import Butler
+from SCons import Action
+
+class callAction(Action.FunctionAction):
+    def __init__(self, ident, func, **kw):
+        self.ident = ident
+        self.func = func
+        Action.FunctionAction.__init__(self, self.func, kw)
+
+    def get_contents(self, *args, **kwargs):
```

Comment:
I should comment this, get_contents is what scons calls to generate a function signature, and should be deterministic. The default get_contents in Action.FunctionAction does not generate consistent signatures across runs of functools.partial, so we overload it to return something deterministic